bc_obps/service/operation_service_v2.py

```diff
@@ -159,3 +167,123 @@ def register_operation_operation_representative(
         operation.contacts.set(contact_ids_to_add_to_operation)
         operation.set_create_or_update(user_guid)
         return operation
+
+    # @classmethod
+    # @transaction.atomic()
+    # def create_registration_purpose(cls, user_guid: UUID, operation_id: UUID, payload: RegistrationPurposeIn) -> Dict[str, Union[str, UUID]]:
+
+    #     registration_payload = RegistrationPurposeIn(
+    #         registration_purpose=payload.registration_purpose, regulated_products=payload.regulated_products
+    #     )
+    #     cls.set_registration_purpose(user_guid, operation_id, registration_payload)
+
+    @classmethod
+    @transaction.atomic()
+    def create_or_update_operation_v2(
+        # brianna better to do as a default arg?
+        cls,
+        user_guid: UUID,
+        operation_id: UUID | None,
+        payload: OperationInformationIn,
+    ) -> Operation:
+        user_operator: UserOperator = UserDataAccessService.get_user_operator_by_user(user_guid)
+
+        operation_data = payload.dict(
+            include={
+                'name',
+                "type",
+                "naics_code_id",
+                'secondary_naics_code_id',
+                'tertiary_naics_code_id',
+            }
+        )
+        operation_data['operator_id'] = user_operator.operator_id
+        if operation_id:
+            operation_data['pk'] = operation_id
+
+        operation: Operation = Operation.custom_update_or_create(Operation, user_guid, **operation_data)
+
+        # set m2m relationships
+        operation.activities.set(payload.activities)
+
+        boundary_map = DocumentService.create_or_replace_operation_document(
+            user_guid, operation.id, payload.boundary_map, 'boundary_map'  # type: ignore # mypy is not aware of the schema validator
+        )
+
+        process_flow_diagram = DocumentService.create_or_replace_operation_document(
+            user_guid, operation.id, payload.boundary_map, 'boundary_map'  # type: ignore # mypy is not aware of the schema validator
+        )
+
+        operation.documents.set([boundary_map, process_flow_diagram])
+
+        # handle multiple operators
+        multiple_operators_data = payload.multiple_operators_array
+        cls.upsert_multiple_operators(operation, multiple_operators_data, user_guid)
+
+        return operation
+
+    @classmethod
+    @transaction.atomic()
+    def register_operation_information(
+        cls, user_guid: UUID, operation_id: UUID | None, payload: OperationInformationIn
+    ) -> Operation:
+        if operation_id:
```

Comment:
Possible tech debt card--a service (or decorator or something) to check external users' permissions. The endpoint decorator is only role-based, and we often need to ensure a user can only see their own operators/operations/facilities, etc.